api/src/main/java/uk/gov/dhsc/htbhf/claimant/entitlement/BackDatedCycleEntitlementCalculator.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.dhsc.htbhf.claimant.entitlement;
+
+import org.springframework.stereotype.Service;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import static java.util.Collections.emptyList;
+import static java.util.Collections.min;
+
+/**
+ * Calculates the number of back dated vouchers a claimant is entitlement to.
+ * The new children that have been matched to the expected due date and the expected due date are used to
+ * calculate the difference between the number of vouchers that were received for pregnancy and
+ * the number of vouchers that should have been received for the new children. If the difference in vouchers
+ * is less than zero, zero is returned.
+ */
+@Service
+public class BackDatedCycleEntitlementCalculator {
+
+    private final Integer entitlementCalculationDurationInDays;
+    private final EntitlementCalculator entitlementCalculator;
+
+    /**
+     * Constructor for {@link BackDatedCycleEntitlementCalculator}.
+     *
+     * @param paymentCycleConfig    configuration for the payment cycle
+     * @param entitlementCalculator calculates entitlement for a single calculation period
+     */
+    public BackDatedCycleEntitlementCalculator(PaymentCycleConfig paymentCycleConfig,
+                                               EntitlementCalculator entitlementCalculator) {
+        this.entitlementCalculationDurationInDays = paymentCycleConfig.getEntitlementCalculationDurationInDays();
+        this.entitlementCalculator = entitlementCalculator;
+    }
+
+    public Integer calculateBackDatedVouchers(Optional<LocalDate> expectedDueDate, List<LocalDate> newChildrenDateOfBirths) {
```

Comment:
I think this needs javadoc to explain what the parameters are

Reply:
Updated.